spec.md

```diff
@@ -233,6 +233,26 @@ circuit:
       node notA = not(a)
 ```
 
+Multiple optional group definitions may reference the same group declaration.
+However, optional group definitions may not reference signals in other definitions, even if their group definitions reference the same group declaration.
```

Comment:
Maybe "a group definition forms a lexical scope for all identifiers, two groups referencing the same group declaration are separate lexical scopes".

Reply:
There's some more precision language earlier that is not as terse as this. (I like this.)

> Declarations of identifiers and references to existing identifiers following the same lexical scoping rules as FIRRTL conditional statements (see: [@sec:conditional-scopes])---identifiers declared in the group definition may not be used outside the group while groups may refer to identifiers declared outside the group.

I was attempting to say, "I shouldn't have to say this because it says the same thing two paragraphs up. However, I want to make sure you know you can't escape lexical scoping this way."

I'll add something closer to your language here.